### PR TITLE
added mermaid diagram

### LINKED ISSUE #524: add mermaid.js diagram
This Issue is an sub-issue of #257

Describe the solution you'd like
Re-add [mermaid.js](https://mermaid-js.github.io/mermaid/#/) diagram type. This can be done via [mermaid](https://www.npmjs.com/package/mermaid) and [@types/mermaid](https://www.npmjs.com/package/@types/mermaid)
the language for code blocks was mermaid in 1.6
Building a react component should not be to complicated as proven by [this](https://github.com/jasonbellamy/react-mermaid/blob/master/src/react-mermaid.js).

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
markdown editor

### Description
This PR readds mermaid diagrams.

### Steps

- [x] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [ ] extended changelog

### Related Issue(s)
closes #524

### CODE REVIEW

**Comment by DerMolly** [comment]
this works so far, but has some problems:
1. [ ] the svg is too wide
grafik
2. [ ] the mermaid.scss file (a straight copy from the server repo) seems to need updating

**Comment by mrdrogdrog** [comment]
I fixed the width problem for you. And now please excuse me. I've got to go and burn my hands.

**Review by mrdrogdrog — APPROVED** [review]

File: src/components/markdown-renderer/replace-components/mermaid/mermaid-chart.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+export const MermaidChart: React.FC<MermaidChartProps> = ({ code }) => {
+  const [diagramId] = useState(() => 'mermaid_' + uuid().replaceAll('-', '_'))
+  const diagramContainer = useRef<HTMLDivElement>(null)
+  const [error, setError] = useState<string>()
+  const { t } = useTranslation()
+
+  useEffect(() => {
+    if (!mermaidInitialized) {
+      mermaid.initialize({ startOnLoad: false })
+      console.log('mermaid init')
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/mermaid/mermaid-chart.tsx:26]: maybe this should be removed?

Comment (mrdrogdrog) [reply src/components/markdown-renderer/replace-components/mermaid/mermaid-chart.tsx]: removed

File: public/locales/en.json [hunk 2]
```diff
@@ -6,6 +6,9 @@
     "renderer": {
         "flowchart": {
             "invalidSyntax": "Invalid flowchart.js syntax!"
+        },
+        "mermaid":         {
```
Comment (ErikMichelson) [inline public/locales/en.json:10]:
```suggestion
        "mermaid": {
```

**Review by ErikMichelson — APPROVED** [review]

**Review by mrdrogdrog — APPROVED** [review]

**Pull request merged**